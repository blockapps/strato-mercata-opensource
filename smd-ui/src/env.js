const http_protocol = (window.IS_SSL && window.IS_SSL === 'true') ? 'https' : 'http';
const ws_protocol = (window.IS_SSL && window.IS_SSL === 'true') ? 'wss' : 'ws';

const apex_uri = '/apex-api';
const bloc_uri = '/bloc/v2.2';
const bloc_doc_uri = '/docs/?url=/bloc/v2.2/swagger.json';
const cirrus_uri = '/cirrus/search';
const strato_uri = '/strato-api/eth/v1.2';
const strato_doc_uri = '/docs/?url=/strato-api/eth/v1.2/swagger.json';
const strato_v23_uri = '/strato/v2.3'
const health_uri = '/health'

const node_host = window.NODE_HOST && window.NODE_HOST !== '__NODE_HOST__' ? window.NODE_HOST : (process.env.REACT_APP_NODE_HOST ? process.env.REACT_APP_NODE_HOST : 'localhost');
const oauth_enabled = window.OAUTH_ENABLED && window.OAUTH_ENABLED !== '__OAUTH_ENABLED__' ? window.OAUTH_ENABLED==='true' : process.env.REACT_APP_OAUTH_ENABLED==='true';

export const env = {
  NODE_NAME: window.NODE_NAME && window.NODE_NAME !== '__NODE_NAME__' ? window.NODE_NAME : 'LOCALHOST',
  APEX_URL: `${http_protocol}://${node_host}${apex_uri}`,
  BLOC_URL: `${http_protocol}://${node_host}${bloc_uri}`,
  BLOC_DOC_URL: `${http_protocol}://${node_host}${bloc_doc_uri}`,
  CIRRUS_URL: `${http_protocol}://${node_host}${cirrus_uri}`,
  HEALTH_URL: `${http_protocol}://${node_host}${health_uri}`,
  NODE_HOST: node_host,
  OAUTH_ENABLED: oauth_enabled,
  STRATO_URL: `${http_protocol}://${node_host}${strato_uri}`,
  STRATO_URL_V23: `${http_protocol}://${node_host}${strato_v23_uri}`,
  STRATO_DOC_URL: `${http_protocol}://${node_host}${strato_doc_uri}`,
  SOCKET_SERVER: `${ws_protocol}://${node_host}/`,
  POLLING_FREQUENCY: window.POLLING_FREQUENCY && window.POLLING_FREQUENCY !== '__POLLING_FREQUENCY__' ? window.POLLING_FREQUENCY : 5 * 1000,
  STRATO_VERSION: window.STRATO_VERSION && window.STRATO_VERSION !== '__STRATO_VERSION__' ? window.STRATO_VERSION : 'N/A',
};

