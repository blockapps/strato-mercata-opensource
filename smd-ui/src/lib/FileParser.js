export function getImportStatements(source) {
    const splittedSource = source.split("\n")
    // TODO: This does not guard against multiline strings or comments
    const newImports = splittedSource.filter((line) => /^\s*import /.test(line))
    const newFileList = []
    newImports.map((item) => {
        const splittedItem = item.split(' ')
        const filename = splittedItem.filter((word) => word.includes('.sol'))
        newFileList.push(filename.length > 0 && filename[0].replace(/['";]+/g, ''))
        return item
    })
    return newFileList
}

export function replaceImportStatementsWithSource(source, importsCode, tabData) {
    const splittedSource = source.split("\n")
    const filteredList = splittedSource.map((item) => {
        const code = importsCode.filter(
            (codeItem) =>
                item.includes(codeItem.title)
        )
        return code.length > 0 ? getFileAndReplaceWithImport(code[0].text, tabData) : item }
    )
    filteredList.join("\r\n")
    return filteredList
}

export function getFileAndReplaceWithImport(code, tabData) {
    const newFileList = getImportStatements(code)
    if (newFileList.length === 0) {
        return code
    }
    const items = tabData.filter((item) =>
        newFileList.includes(item.title)
    )
    const replacedCode = replaceImportStatementsWithSource(code, items, tabData)
    return replacedCode.join("\r\n")
}
