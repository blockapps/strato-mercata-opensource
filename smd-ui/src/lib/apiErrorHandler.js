export function handleApiError(response) {
  return new Promise(function(resolve,reject){
    if(response.error) {
      reject(response.error);
    }
    else {
      resolve(response);
    }
  })
}
