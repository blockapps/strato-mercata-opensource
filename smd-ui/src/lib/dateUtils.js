import moment from 'moment';

export function parseDateFromString(timeAsString) {
  const utcDate = timeAsString.replace(' UTC', 'Z')
  const formattedDate = moment(utcDate).format('YYYY-MM-DD hh:mm:ss A')
  return formattedDate
}

export function parseDateFromTimestamp(timestamp) {
  return parseDateFromString(moment.unix(timestamp).utc().toString());
}
