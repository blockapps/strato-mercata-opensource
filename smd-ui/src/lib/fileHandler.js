export function downloadFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(content));
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}

export function downloadPDFFile(name, content) {
    var element = document.createElement('a');
    element.setAttribute('href', content);
    element.setAttribute('download', name);
    element.style.display = 'none';
    document.body.appendChild(element);
    element.click();
    document.body.removeChild(element);
}