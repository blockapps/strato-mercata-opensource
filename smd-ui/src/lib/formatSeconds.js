import React from "react";
export function sec2Date(sec){
    if (sec === undefined || sec <= 0 ){
        return <div></div>;
    }
    let seconds = sec;
    const days = Math.floor(seconds / (3600*24));
    const daysString = days.toString().padStart(2, "0");
    seconds  -= days*3600*24;
    const hrs = Math.floor(seconds / 3600);
    const hoursString = hrs.toString().padStart(2, "0");
    seconds  -= hrs*3600;
    const mnts = Math.floor(seconds / 60);
    const minutesString = mnts.toString().padStart(2, "0");
    seconds  -= mnts*60;
    const secondsString = Math.floor(seconds).toString().padStart(2, "0");
    const ret = <div>
        Uptime {days > 0 ? <span>{daysString}d:</span> : null}
         <span>
             {hoursString}:
        </span>
        <span>
             {minutesString}:

        </span>
        <span>
             {secondsString}
        </span>
    </div>
    return ret;
}
