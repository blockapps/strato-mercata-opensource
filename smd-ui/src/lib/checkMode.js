import { env } from "../env";

export function isOauthEnabled() {
  return env.OAUTH_ENABLED
}