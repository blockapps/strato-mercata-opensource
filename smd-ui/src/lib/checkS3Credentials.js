import { env } from "../env";

export function isS3Available() {
  return env.EXT_STORAGE_ENABLED === 'true'
}
