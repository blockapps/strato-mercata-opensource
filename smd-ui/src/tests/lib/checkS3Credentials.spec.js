import { isS3Available } from '../../lib/checkS3Credentials';
import * as enviorment from "../../env";

describe('Lib: checkS3Credentials', () => {

  test('when s3 is unavailable', () => {
    enviorment.env.EXT_STORAGE_ENABLED = 'false';
    expect(isS3Available()).toMatchSnapshot();
  });

  test('when s3 is available', () => {
    enviorment.env.EXT_STORAGE_ENABLED = 'true';
    expect(isS3Available()).toMatchSnapshot();
  });
});
