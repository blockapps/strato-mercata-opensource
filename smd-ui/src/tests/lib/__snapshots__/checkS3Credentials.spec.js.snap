// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lib: checkS3Credentials when s3 is available 1`] = `true`;

exports[`Lib: checkS3Credentials when s3 is unavailable 1`] = `false`;
